src/resources/auth/middleware/Permit.ts

```diff
@@ -1,18 +1,77 @@
 import express from "express";
-import { User } from "../../../generated/models/User.generated";
 import { PermissionFlag } from "./PermissionFlag";
+import { checkPermissionForRole } from "./Roles";
+import { AuthUser } from "../../../generated/models/AuthUser.generated";
+import { PermissionFilter } from "../filter/PermissionFilter";
+import { ResourcePermissionController } from "../controllers/ResourcePermissionController";
 
 export class Permit {
+	static authorizesToManipulateResource =
+		(resourceController: ResourcePermissionController) =>
+		(req: express.Request, res: express.Response, next: express.NextFunction) => {
+			const identifier = req.params.identifier;
+			const authUser = req.user as AuthUser;
+
+			if (!req.user || !identifier || !authUser || !authUser.organizationIdentifier) {
+				res.status(403).send();
+				return;
+			}
+
+			const itIsAboutTheOrganizationOfTheUser = authUser.organizationIdentifier === identifier;
+			if (itIsAboutTheOrganizationOfTheUser) {
+				next();
+				return;
+			}
+
+			const permissionFilter = PermissionFilter.buildOwnershipPermissionFilter(
+				identifier,
+				authUser.organizationIdentifier,
+			);
+
+			(async () => {
+				try {
+					const isExist = await resourceController.isExist(permissionFilter);
+
+					if (!isExist) {
+						res.status(403).send();
+						return;
+					}
+					next();
+				} catch (error) {
+					res.status(500).send();
+				}
+			})();
+		};
+
+	static authorizesForAction = () => (req: express.Request, res: express.Response, next: express.NextFunction) => {
+		const action = req.method + ":" + req.route.path;
+		if (!req.user) {
+			res.status(403).send();
+			return;
+		}
+		const member: AuthUser = req.user as AuthUser;
+
+		if (checkPermissionForRole(member.role, action)) {
```

Comment:
Hier könnten wir mit dem Namen etwas konsistenter sein:

```suggestion
		const authUser = req.user as AuthUser;

		if (checkPermissionForRole(authUser.role, action)) {
```

Weiter unten wird auch `u` als Variablenname benutzt – ich bin aber kein großer Fan von Ein-Buchstaben-Variablen. :P

Reply:
Habe beides jetzt user genannt. member sollte ausdrücken, in welcher Funktion der User gerade nach seiner Rolle gefragt wird. Es ist sozusagen nicht die rolle für die Kulturdatenbank, sondern die Rolle als Mitglied in einer Organisation. Ist aber wohl eher verwirrend hier.